src/libraries/System.Private.CoreLib/src/System/Globalization/CompareInfo.Windows.cs

```diff
@@ -129,18 +129,24 @@ private unsafe int GetHashCodeOfStringCore(ReadOnlySpan<char> source, CompareOpt
             Debug.Assert(!GlobalizationMode.Invariant);
             Debug.Assert((options & (CompareOptions.Ordinal | CompareOptions.OrdinalIgnoreCase)) == 0);
 
-            if (source.Length == 0)
```

Comment:
Could you please check if Windows/ICU returns the same hash value for the empty string with all locales or cultures? I am asking because if this is the case, we can cache this value and keep the fast path for the empty strings.

Reply:
By the way, I am not actively following up so either don't block on me or wait till I am back. Thanks!

Reply:
Or at least maybe we could cache it for invariant culture (if not already)?

Reply:
I thought about this but currently I don't have any evidence that anybody is trying to compute the culture-aware hash code of an empty string in a hot path.

And yeah, no rush whatsoever on this. It can wait until the new year. :)